src/app/cases/case_components/case-follow-up-add-visit/case-follow-up-add-visit.component.ts

```diff
@@ -0,0 +1,145 @@
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ADD_EDIT_FORM_ID } from '../../../app.constants';
+import { FormBase } from '../../../shared/dynamic-form/types/form-element-base';
+import { VisitService } from '../../../_services/api/visit.service';
+import { FormElementControlService } from '../../../_services/form-element-control.service';
+import * as data from './case-follow-up-add-visit-data';
+
+@Component({
+  selector: 'app-case-follow-up-add-visit',
+  templateUrl: './case-follow-up-add-visit.component.html',
+  styleUrls: ['./case-follow-up-add-visit.component.scss'],
+})
+export class CaseFollowUpAddVisitComponent implements OnInit, OnDestroy {
+  myFormElements: FormBase<any>[] = data.FORM_DATA_CASE_FOLLOW_UP_NEW_VISITS;
+  formId = ADD_EDIT_FORM_ID;
+  subscriptions: Subscription[] = [];
+
+  @ViewChild('form') dynamicForm: any;
+
+  constructor(
+    public visitService: VisitService,
+    private formElementControlService: FormElementControlService
+  ) {}
+
+  ngOnInit(): void {
+    this.myFormElements = JSON.parse(JSON.stringify(data.FORM_DATA_CASE_FOLLOW_UP_NEW_VISITS));
+
+    setTimeout(() => {
+      const { form } = this.dynamicForm;
+      if (form) {
+        const availability = form.get('availability');
+        availability.setValue('COOPERATIVE');
+
+        this.subscriptions.push(
+          availability.valueChanges.subscribe((val: string) => {
+            if (val !== 'COOPERATIVE') {
```

Comment:
this if/else statement is huge and hard to follow;
one way to make it more readable could be:
- make a separate method and call it in the subscribe: availability.valueChanges.subscribe((val: string) => this.toggleSections(val === 'COOPERATIVE'))
- in the toggleSections method have an array of strings where you declare the ids of sections that needs to be updated
- iterate throught that update the group attribute (with the boolean value passed to the method)

Reply:
Simplified it.